hamster_gtk/tracking/screens.py

```diff
@@ -178,13 +188,105 @@ def __init__(self, controller, *args, **kwargs):
 
         # Fact entry field
         self.raw_fact_entry = Gtk.Entry()
+        self.raw_fact_entry.set_completion(self._get_completion())
         self.pack_start(self.raw_fact_entry, False, False, 0)
 
         # Buttons
         start_button = Gtk.Button(label=_('Start Tracking'))
         start_button.connect('clicked', self._on_start_tracking_button)
         self.pack_start(start_button, False, False, 0)
 
+    def _get_completion(self):
+        """
+        Return a completion instance to match 'activity@category' strings.
+
+        Returns:
+            Gtk.EntryCompletion: Completion instance.
+        """
+        def remove_duplicates(list):
+            """
+            Quick and dirty function to remove duplicate list items.
+
+            The first item encounterd is keept, the following thrown away.
+
+            We can not simply turn this into a set as we would loose the
+            ordering.  This clearly is highly inefficiennt and the only reason
+            we do this for now is the very limited size of the input list.
+
+            For one possible way to do this see: ``https://code.activestate.com/recipes/576694/``.
+            """
+            result = []
+            for item in list:
+                if item not in result:
+                    result.append(item)
+            return result
+
+        def match_anywhere(completion, entrystr, iter, data):
+            """
+            Check if the entrysring is a substring of our text_column.
+
+            Args:
+                completion (Gtk.EntryCompletion): The completion instance.
+                entrystr (str): The text extracted from the entry so far.
+                iter: Iterator position indicating the model row we check against.
+                data: ???.
```

Comment:
This is the `None` passed to the `set_match_func` method: https://lazka.github.io/pgi-docs/#Gtk-3.0/callbacks.html#Gtk.EntryCompletionMatchFunc